pkg/api/org_users.go

```diff
@@ -211,21 +212,34 @@ func (hs *HTTPServer) getOrgUsersHelper(c *models.ReqContext, query *org.GetOrgU
 
 	filteredUsers := make([]*org.OrgUserDTO, 0, len(result))
 	userIDs := map[string]bool{}
+	authLabelsUserIDs := make([]int64, 0, len(result))
 	for _, user := range result {
 		if dtos.IsHiddenUser(user.Login, signedInUser, hs.Cfg) {
 			continue
 		}
 		user.AvatarURL = dtos.GetGravatarUrl(user.Email)
 
 		userIDs[fmt.Sprint(user.UserID)] = true
+		authLabelsUserIDs = append(authLabelsUserIDs, user.UserID)
 		filteredUsers = append(filteredUsers, user)
 	}
 
-	// Get accesscontrol metadata for users in the target org
+	modules, err := hs.authInfoService.GetUserLabels(c.Req.Context(), models.GetUserLabelsQuery{
+		UserIDs: authLabelsUserIDs,
+	})
+
+	if err != nil {
+		hs.log.Warn("failed to retrieve users IDP label", err)
+	}
+
+	// Get accesscontrol metadata and IPD labels for users in the target org
 	accessControlMetadata := hs.getMultiAccessControlMetadata(c, query.OrgID, "users:id:", userIDs)
 	if len(accessControlMetadata) > 0 {
```

Comment:
Now we just need to remove this check otherwise we will not append the labels if we don't have any access control metadata and that could happen if the request is missing the accesscontrol=true query string

Reply:
Do you mean like this?

```suggestion
	// Get accesscontrol metadata and IPD labels for users in the target org
	accessControlMetadata := hs.getMultiAccessControlMetadata(c, query.OrgID, "users:id:", userIDs)
	if len(accessControlMetadata) > 0 {
		for i := range filteredUsers {
			filteredUsers[i].AccessControl = accessControlMetadata[fmt.Sprint(filteredUsers[i].UserID)]
			filteredUsers[i].AuthLabels = []string{login.GetAuthProviderLabel(modules[filteredUsers[i].UserID])}
		}
	}
```

Reply:
I think you meant like this.

```
	// Get accesscontrol metadata and IPD labels for users in the target org
	accessControlMetadata := hs.getMultiAccessControlMetadata(c, query.OrgID, "users:id:", userIDs)
	for i := range filteredUsers {
		filteredUsers[i].AccessControl = accessControlMetadata[fmt.Sprint(filteredUsers[i].UserID)]
		if module, ok := modules[filteredUsers[i].UserID]; ok {
			filteredUsers[i].AuthLabels = []string{login.GetAuthProviderLabel(module)}
		}
	}```